### DISCUSSION TITLE
How do I clip an image with a circular mask?

### DISCUSSION THREAD

**Discussion by dyc3** [typst/typst#2641] [Q&A] [answered]
I'm working on porting my resume from latex to typst.

I have an image that is a square:

![image](https://github.com/typst/typst/assets/1808807/1de328f3-ffc7-44fe-a24d-33ad2b96c41b)

I want to mask it with a circle with a border around it, like this:

![image](https://github.com/typst/typst/assets/1808807/ad02a3f1-4690-4edc-aa05-149c6d77a671)

I can't figure out how to do this with typst's visualization functions. Any ideas?

**Comment by johannes-wolf** [comment, marked as answer]
You can use `box` or `block` with `clip: true` and `radius: <width/2>` like this:
```typst
#box(clip: true, stroke: 5pt + yellow, radius: 2.5cm,
  width: 5cm, height: 5cm,
  image("face.png", height: 5cm))
```

![grafik](https://github.com/typst/typst/assets/519002/e524d9bb-6a28-4b2b-85a5-50c5274c6074)

**Reply by laurmaedje** [reply]
Looks like the clipping isn't anti-aliased, that's unfortunate... Is that from Typst raster renderer or PDF?

**Reply by johannes-wolf** [reply]
Its from the raster renderer. PDF looks fine.

**Reply by alkeryn** [reply]
hey thank you for your reply that was what i was looking for, though now i'm curious what if you want to use weirder shapes for your mask let's say a star or an hexagon, do you have an idea how you would do it ?
thank you ! :)

**Reply by royfrancis** [reply]
@johannes-wolf Thanks for this example! I noticed that it doesn't quite work if the aspect ratio of the image is further away from a square.

**Reply by alkeryn** [reply]
@royfrancis you have to do some math, you basically want to take into account the smallest dimension.
